packages/chart/src/components/LinearChart.tsx

```diff
@@ -241,8 +241,10 @@ const LinearChart = forwardRef<SVGAElement, LinearChartProps>(({ parentHeight, p
   })
   // get the number of months between the first and last date
   const dateSpanMonths =
-    [0, data.length - 1].map(i => parseDate(data[i][runtime.xAxis.dataKey])).reduce((a, b) => Math.abs(a - b), 0) /
-    (1000 * 60 * 60 * 24 * 30)
+    data.length && isDateTime
+      ? [0, data.length - 1].map(i => parseDate(data[i][runtime.xAxis.dataKey])).reduce((a, b) => Math.abs(a - b), 0) /
+        (1000 * 60 * 60 * 24 * 30)
+      : 0
```

Comment:
Great point! I cleaned it up a bit! The reducer is because there are 2 values, the first and last date - and then it is calculating the difference before converting it into months